ciao-deploy/cmd/unjoin.go

```diff
@@ -0,0 +1,67 @@
+// Copyright © 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"os/user"
+	"syscall"
+
+	"github.com/01org/ciao/ciao-deploy/deploy"
+	"github.com/spf13/cobra"
+)
+
+// unjoinCmd represents the unjoin command
+var unjoinCmd = &cobra.Command{
+	Use:   "unjoin <hosts>",
+	Short: "Remove the specified nodes from the cluster",
+	Long: `Remove the nodes from the cluster. Removing certificates and
+	 uninstalling software.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx, cancelFunc := context.WithCancel(context.Background())
+		defer cancelFunc()
+
+		sigCh := make(chan os.Signal, 1)
+		go func() {
```

Comment:
Great idea. I handled closing the goroutine by selecting on the ctx.Done() channel.